dev/RadialGradientBrush/RadialGradientBrush.cpp

```diff
@@ -230,6 +239,29 @@ void RadialGradientBrush::UpdateCompositionInterpolationSpace()
     }
 }
 
+void RadialGradientBrush::UpdateCompositionExtendMode()
+{
+    MUX_ASSERT(SharedHelpers::IsCompositionRadialGradientBrushAvailable());
```

Comment:
>IsCompositionRadialGradientBrushAvailable [](start = 30, length = 41)

This call is actually pretty expensive and the if statement below does almost the same thing. Can we remove the assert?

Reply:
Are asserts not disabled for release mode?

Reply:
They are, I guess I am more concerned with the call at the reference to this function that isn't in an assert.

---
In reply to: [372534902](https://github.com/microsoft/microsoft-ui-xaml/pull/1903#discussion_r372534902) [](ancestors = 372534902)

Reply:
Wasn't sure how the codebase calibrates between perf and defensiveness - 

I could remove all casting and API availability checks, and instead just cache a reference to the CompositionRadialGradientBrush and rely on null checks against that for logic?

Reply:
I think that is better, but maybe @ranjeshj thinks otherwise.

---
In reply to: [373656794](https://github.com/microsoft/microsoft-ui-xaml/pull/1903#discussion_r373656794) [](ancestors = 373656794)

Reply:
Asserts will be removed in release, so that is not a concern. We could cache the check or use a null check to avoid repeatedly paying that cost.

---
In reply to: [374236134](https://github.com/microsoft/microsoft-ui-xaml/pull/1903#discussion_r374236134) [](ancestors = 374236134,373656794)